tests/trainer/test_dataloaders.py

```diff
@@ -295,6 +295,18 @@ def test_train_inf_dataloader_error(tmpdir):
         trainer.fit(model)
 
 
+@pytest.mark.skip('TODO: speed up this test')
```

Comment:
This has to be resolved before merge

Reply:
Since the "CustomNotImplementedErrorDataloader" is a version of 'CustomInfDataloader' with a minor change the behavior should be very similar.

The tests derived by copy and paste of the corresponding tests for 'CustomInfDataloader'.  The decorators were copied under the assumption that they are correct or useful. Given that I am not sure about them, it makes sense to remove them. @Borda  Please advise if that is a reasonable solution.

On a different note alternatively CustomNotImplementedErrorDataloader could be derived from CustomInfDataloader with the __len__ method  added. This would reduce duplicated code. What do you think?

Reply:
Thanks @Borda After removing the decorators pytest failed. This is because I also changed the match string from "infinite DataLoader" to "not_implemented_error DataLoader" when I copied the test. This does not match the MisconfigurationException exception string raised from pytorch_lightning/trainer/data_loading.py. I am changing the match strings to "infinite DataLoader" and comment skipping the tests out.

Reply:
@Borda I created a new PR #2307 with the changes to make the tests work. You will see a lot of commits there. It seems difficult for the test to pass on all machines. It seems that the machines cancel if it takes more than 15v minutes total time. When I enabled all the tests sometimes they finished in time on a few machine, but never on all of them.  On my laptop I have observed when running the local test that they sometimes just hang (a single test >15 minutes). Maybe this is a problem of pytest in my environment (MacOS), or a more general issue. This merge certainly fixes the issue I observed, and the test in the new PR are technically working. I suggest to continue with the commits of the new PR.